packages/pyright-internal/src/tests/workspaceEditUtils.test.ts

```diff
@@ -46,6 +48,99 @@ test('test applyWorkspaceEdits changes', async () => {
     assert.strictEqual(cloned.test_program.getSourceFile(range.fileName)?.getFileContent(), 'Text Changed');
 });
 
+test('test edit mode for workspace', async () => {
```

Comment:
I think we need to have tests that changes same file multiple times and has nested changes such as

First add new import statement `from test import a, c` and then add new symbol to the new import you added in `from test import a[, b], c` and then remove import statement or reorder import statement in the file (assuming it had existing import statement)

or you can think about other multiple edits, nested edit case

let's say we provide refactoring to merge `import statements` and `remove unused imports`. so user doesn't have duplicated import statements or same symbol with different name such as `import os as o` and `import os as os` and etc.

Reply:
I changed the test to do a nested change. I didn't think it would matter if there was more than one, so I just added one extra nested change and verified we ended up with a single change.